src/app/modules/submission/submission.utils.ts

```diff
@@ -125,8 +117,14 @@ export const getFilteredResponses = (
   form: IFormDocument,
   responses: FieldResponse[],
 ): Result<FilteredResponse[], ConflictError> => {
-  const responseModeFilter = getResponseModeFilter(form.responseMode)
-  const formFieldModeFilter = getFormFieldModeFilter(form.responseMode)
+  const responseModeFilter = getResponseModeFilter(
+    form.responseMode === FormResponseMode.Encrypt &&
+      !form.get('newEncryptionBoundary'),
+  )
+  const formFieldModeFilter = getFormFieldModeFilter(
+    form.responseMode === FormResponseMode.Encrypt &&
+      !form.get('newEncryptionBoundary'),
```

Comment:
consider: 

const isEncryptedMode (or isE2EEMode?) = FormResponseMode.Encrypt && !form.get('newEncryptionBoundary')

const responseModeFilter = getResponseModeFilter(isEncryptedMode)
const formFieldModeFilter = getFormFieldModeFilter(isEncryptedMode)

Reply:
done in [3b8025a](https://github.com/opengovsg/FormSG/pull/6595/commits/3b8025a70513f8472ad4e6f00d205f32ba2fe5b7), thanks!